### ISSUE TITLE
Make the `< result >` row be optional for Context Expressions on DMN Editor's Boxed Expression Editor

### ISSUE THREAD

**Issue by tiagobento** [apache/incubator-kie-tools#3748]
```[tasklist]
### Acceptance criteria
- [ ] Write a nice description for this issue.
- [ ] Add acceptance criteria here.
```

**Comment by jomarko** [comment]
@tiagobento thank you for this issue. From what I know, it is kind of 'optional' already. meaning that if user do not fill this `<result>` row with any text/data, then dmn engine returns all entries of nearest parent context box.

Is this kie-issue about not adding `<result>` row by default? Is your idea to add this `<result>` row just in case user explicitly want it? For example context menu should contain and menu item `insert <result> row`?

**Comment by tiagobento** [comment]
@jomarko Yeah, but the UI doesn't reflect that very well.. Users might be confused that an empty `"<result>"` row with the "Select expression" indication means that there's something missing there. Another thing to consider is that having or not having the `"<result>"` row will drastically change the behavior of a Context boxed expression..

_"Is this kie-issue about not adding <result> row by default?"_
Not necessarily... I think it's important that users know that there's an obvious option to add a `"<result>"` row if they want.

_"Is your idea to add this <result> row just in case user explicitly want it?"_
Yes

_"For example context menu should contain and menu item insert <result> row?"_
Maybe... but I think that would be too hidden... We'll need to explore UI options for this, without losing the indication that the `"<result>"` row option is avaible.
